server/channels/app/notification.go

```diff
@@ -121,35 +121,10 @@ func (a *App) SendNotifications(c request.CTX, post *model.Post, team *model.Tea
 		groups = result.Data.(map[string]*model.Group)
 	}
 
-	mentions := &ExplicitMentions{}
-	allActivityPushUserIds := []string{}
-	var allowChannelMentions bool
-	var keywords map[string][]string
-	if channel.Type == model.ChannelTypeDirect {
-		otherUserId := channel.GetOtherUserIdForDM(post.UserId)
-
-		_, ok := profileMap[otherUserId]
-		if ok {
-			mentions.addMention(otherUserId, DMMention)
-		}
-
-		if post.GetProp("from_webhook") == "true" {
-			mentions.addMention(post.UserId, DMMention)
-		}
-	} else {
-		allowChannelMentions = a.allowChannelMentions(c, post, len(profileMap))
-		keywords = a.getMentionKeywordsInChannel(profileMap, allowChannelMentions, channelMemberNotifyPropsMap)
-
-		mentions = getExplicitMentions(post, keywords, groups)
-		// Add an implicit mention when a user is added to a channel
-		// even if the user has set 'username mentions' to false in account settings.
-		if post.Type == model.PostTypeAddToChannel {
-			addedUserId, ok := post.GetProp(model.PostPropsAddedUserId).(string)
-			if ok {
-				mentions.addMention(addedUserId, KeywordMention)
-			}
-		}
+	mentions, keywords := a.getExplicitMentionsAndKeywords(c, post, channel, profileMap, groups, channelMemberNotifyPropsMap, parentPostList)
 
+	var allActivityPushUserIds []string
+	if channel.Type != model.ChannelTypeDirect {
```

Comment:
Why group mentions stay outside of the `getExplicitMentionsAndKeywords` function?

Reply:
This loop checks if all members mentioned as part of a user group are missing in the channel, if so, sends a warning message to a user. We don't wanna do this when deleting a message. Because the `getExplicitMentionsAndKeywords` function is called both when creating a post and deleting, we keep this loop outside of `getExplicitMentionsAndKeywords` so it can be checked only when creating the post.